vuu-ui/packages/vuu-utils/src/logging-utils.ts

```diff
@@ -0,0 +1,76 @@
+export interface LogFn {
+	(message?: unknown, ...optionalParams: unknown[]): void;
+}
+export interface AssertLogFn {
+	(condition: boolean, message?: unknown, errorMessaage?: unknown): void;
+}
+
+export interface TableLogFn {
+	(properties?: object): void
+}
+
+export interface Logger {
+	log: LogFn;
+	warn: LogFn;
+	error: LogFn;
+	group: LogFn;
+	groupCollapsed: LogFn;
+	groupEnd: LogFn;
+	assert: AssertLogFn;
+	trace: LogFn;
+	debug: LogFn;
+	info: LogFn;
+	table: TableLogFn
+}
+
+export type BuildEnv = 'production' | 'development';
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const NO_OP: LogFn = () => {};
+
+export class ConsoleLogger implements Logger {
+	readonly log: LogFn;
+	readonly warn: LogFn;
+	readonly error: LogFn;
+	readonly group: LogFn;
+	readonly groupCollapsed: LogFn;
+	readonly groupEnd: LogFn;
+	readonly assert: AssertLogFn;
+	readonly trace: LogFn;
+	readonly debug: LogFn;
+	readonly info: LogFn;
+	readonly table: TableLogFn;
+
+	constructor(options?: { buildEnv?: string }, level?: string) {
+		const { buildEnv } = options || {};
+
+		if (buildEnv === 'production') {
```

Comment:
I'm not sure about this approach tbh.  Firstly, even having NO_OP method calls in performance sensitive sections a code worries me a bit. It may be that the javascript runtime is smart enough to optimise them away entirely, maybe not. But then any runtime arguments are definitiely going to be evaluated, whether the function does anything or not. 
There may be cases where we want to build a complex structured message, maybe just a JSON .stringify - in these cases we will still be occurring that cost even when the log method is a noop.

What I was thinking was that we would wrap the actual logging statements with environment checks, using substituted environment variables, so that in a prod build, there is no logging code at all in the source (the minifier would eliminate it)